src/api/fungibleAsset.ts

```diff
@@ -46,6 +46,29 @@ export class FungibleAsset {
     return getFungibleAssetMetadata({ aptosConfig: this.config, ...args });
   }
 
+  /**
+   * Queries the current specific fungible asset metadata
+   *
+   * This query returns the fungible asset metadata for a specific fungible asset.
+   *
+   * @param assetType The asset type of the fungible asset.
+   * e.g
+   * "0x1::aptos_coin::AptosCoin" for Aptos Coin
+   * "0xc2948283c2ce03aafbb294821de7ee684b06116bb378ab614fa2de07a99355a8" - address format if this is fungible asset
+   *
+   * @returns getFungibleAssetMetadata A fungible asset metadata item
+   */
+  async getFungibleAssetMetadataByAssetType(assetType: string): Promise<GetFungibleAssetMetadataResponse> {
+    return getFungibleAssetMetadata({
```

Comment:
So look like you return an array even though we explicitly query for an item. I think it makes more sense to return and item, and if it is not found it would simply resolve to `undefined` 
```
async getFungibleAssetMetadataByAssetType(assetType: string): Promise<GetFungibleAssetMetadataResponse[0]> {
    const res = await getFungibleAssetMetadata({
      aptosConfig: this.config,
      options: {
        where: {
          asset_type: { _eq: assetType },
        },
      },
    });
    return res[0];
  }
```

Reply:
Updated.